plugins/experimental/traffic_dump/traffic_dump.cc

```diff
@@ -236,6 +242,43 @@ write_content_node(int64_t num_body_bytes)
   return std::string(R"(,"content":{"encoding":"plain","size":)" + std::to_string(num_body_bytes) + '}');
 }
 
+/** Initialize the default cookie to be dumped instead of the cookie seen on the wire.
+ */
+void
+initialize_default_cookie()
+{
+  constexpr size_t default_cookie_size = 2048;
+  default_cookie.reserve(default_cookie_size);
+
+  for (auto i = 0u; i < default_cookie_size; i += 8) {
+    char iteration_buffer[9]; // 7 character number + 1 space + 1 null terminator.
+    sprintf(iteration_buffer, "%07x ", i / 8);
+    default_cookie.append(iteration_buffer);
+  }
+}
+
+/** Inspect the field for any modification traffic_dump has to do for it.
+ *
+ * @param[in] name The field name to inspect.
+ * @param[in] original_value The field value to inspect.
+ *
+ * @return The value traffic_dump should dump for the given field.
+ */
+std::string_view
+filter_fields(std::string_view name, ts::TextView original_value)
+{
+  if (!dump_raw_cookies && (strcasecmp(name, "Set-Cookie") == 0 || strcasecmp(name, "Cookie") == 0)) {
+    auto new_cookie_size = original_value.size();
+    if (original_value.size() > default_cookie.size()) {
+      new_cookie_size = default_cookie.size();
+      TSError("[%s] Encountered a cookie larger than our default cookie size. Default size: %zu, incoming cookie size: %zu",
+              PLUGIN_NAME, default_cookie.size(), original_value.size());
+    }
+    return ts::TextView{default_cookie.data(), new_cookie_size};
+  }
+  return original_value;
+}
```

Comment:
It seems like the use of ts::TextView in this function should just be std::string_view.

Reply:
Good catch. I was going to use the prefix functionality, but I wound up not needing that.